kong/plugins/zipkin/opentracing.lua

```diff
@@ -48,23 +46,21 @@ end
 
 function OpenTracingHandler:initialise_request(conf, ctx)
 	local tracer = self:get_tracer(conf)
-	local var = ngx.var
-	local wire_context = tracer:extract("http_headers", ngx.req.get_headers()) -- could be nil
-	local method, request_url -- if reading request line fails then var.request_uri is nil
-	if var.request_uri then
```

Comment:
For the sake of clarity, James did not "mention to me" that he encountered the problem, I deduced as much from looking at his code + comment.

Reply:
> I am very curious; what were the circumstances under which this was nil? I am very curious to find out. I think we are talking about the Nginx-produced errors that result in error_page being triggered before the rewrite module gets executed (thus Nginx variables are uninitialized).

I triggered it when I sent a non http request to a http port. I can't recall if it was just jibberish sent via netcat, or if it was e.g. a https request to a http port (or a http request to a https port...).

As you noted, `get_method` always results in `"GET"` in this case, but `request_uri` was nil.

Reply:
Yeah, that's the scenario I described above then. Ok, in this case, we should be covered by the PDK changes by @kikito from b0b283897de8f476c6b102ddf8437d238925211f.

Reply:
@thibaultcha see https://github.com/Kong/kong-plugin-zipkin/pull/24#discussion_r226136416

In the case of a non-HTTP request, we don't want to send anything to zipkin for the `http.url` tag. The change in the pdk you linked doesn't help distinguish the case; infact it makes it harder.

Reply:
Also see my comment here: https://github.com/Kong/kong/commit/b0b283897de8f476c6b102ddf8437d238925211f#commitcomment-30949329

Reply:
I think I found the explanation.

The culprint is that kong still executes when an erroneous request is passed. But it executes on the special "error handler" mode, which means that for the erroneous request, it will execute `header_filter`, `body_filter` and `log`.

I have reproduced the situation and checked that when it happens, `kong.request.get_path_with_query` returns an empty string, and not nil. You can find it (with more explanations) here:

https://github.com/Kong/kong/pull/3868

Reply:
Yes, that is the scenario I was referring to above. I think @kikito's solution is elegant enough. His test case on the PDK will help preventing regressions in this API, so that it always returns an empty string in such cases. Ultimately, I think a PDK API is warranted here, to detect such invalid HTTP requests.